### PR TITLE
Set initial default shared preferences

### PR DESCRIPTION
**Pull request by dangeross**
This PR changes the `PaymentOptionsBloc` to set the initial default shared preferences (in iOS also to the app group UserDefaults) if they are currently not stored. This allows the notification service extension to get the default values from shared preferences.

### CODE REVIEW

File: lib/bloc/payment_options/payment_options_bloc.dart [hunk 1]
```diff
@@ -29,22 +29,26 @@ class PaymentOptionsBloc extends Cubit<PaymentOptionsState> {
...
+      } catch (e) {
+        _log.severe("Failed to reset payment options.");
+        rethrow;
+      }
     }
+
+    await getPaymentOptions();
   }
 
   Future<void> getPaymentOptions() async {
```
Comment (erdemyerebasmaz) [inline lib/bloc/payment_options/payment_options_bloc.dart:54]: `resetPaymentOptions` is being called from a UI widget but we can make `getPaymentOptions` a private method as it's not used elsewhere.

File: lib/bloc/payment_options/payment_options_bloc.dart [hunk 2]
```diff
@@ -11,7 +11,7 @@ class PaymentOptionsBloc extends Cubit<PaymentOptionsState> {
   PaymentOptionsBloc(
     this._preferences,
   ) : super(const PaymentOptionsState.initial()) {
-    getPaymentOptions();
+    resetPaymentOptions(true);
   }
```
Comment (erdemyerebasmaz) [inline lib/bloc/payment_options/payment_options_bloc.dart:15]: Realized we can simplify this whole logic to:

```suggestion
        _initializePaymentOptions();
  }

  void _initializePaymentOptions() async {
    await _preferences.hasPaymentOptions() ? await getPaymentOptions() : await resetPaymentOptions();
  }
```

File: ios/Runner.xcodeproj/project.pbxproj
Comment (erdemyerebasmaz) [inline ios/Runner.xcodeproj/project.pbxproj:1]: Why were these changes included? Are they necessary?

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
Have one suggestion... We made it more complex than it is :)

File: lib/bloc/payment_options/payment_options_bloc.dart [hunk 3]
```diff
@@ -29,22 +29,26 @@ class PaymentOptionsBloc extends Cubit<PaymentOptionsState> {
...
-      _log.info(
-        "Reverted payment options to default values: "
-        "proportionalFee: $kDefaultProportionalFee "
-        "exemptFeeMsat: $kDefaultExemptFeeMsat "
-        "channelFeeLimitMsat: $kDefaultChannelSetupFeeLimitMsat",
-      );
-    } catch (e) {
-      _log.severe("Failed to reset payment options.");
-      rethrow;
+  Future<void> resetPaymentOptions([bool checkPaymentOptions = false]) async {
```
Comment (erdemyerebasmaz) [inline lib/bloc/payment_options/payment_options_bloc.dart:32]: `resetPaymentOptions` can be renamed to `setDefaultPaymentOptions` as we're not really resetting anything when shared preferences does not have payment options.

File: ios/Runner.xcodeproj/project.pbxproj
Comment (dangeross) [reply ios/Runner.xcodeproj/project.pbxproj:1]: It wasn't building without adding `ServiceConfig` to the compile sources

File: lib/bloc/payment_options/payment_options_bloc.dart (hunk 2 above)
Comment (dangeross) [reply lib/bloc/payment_options/payment_options_bloc.dart]: Looks a lot simpler now

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM
